sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -2663,6 +2663,36 @@ pi_result cuda_piEnqueueKernelLaunch(
   } catch (pi_result err) {
     retError = err;
   }
+  if (retError == PI_INVALID_VALUE) {
+    // There are no error codes for invalid number of work groups, but we can
+    // print more info
+    CUdevice device = command_queue->device_->get();
+    int maxGridSize[3];
+    if (cuDeviceGetAttribute(&maxGridSize[0],
+                             CU_DEVICE_ATTRIBUTE_MAX_GRID_DIM_X,
+                             device) == CUDA_SUCCESS &&
+        blocksPerGrid[0] > maxGridSize[0]) {
+      std::cerr << "Number of work groups in dimension " << work_dim - 1 << " ("
+                << blocksPerGrid[0] << ") exceeds CUDA maximum X grid size ("
```

Comment:
While it can be queried per device, it currently is the same for all devices (see https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#compute-capabilities=). I can still change that so it will still make sense for if this numbers are changed for future compute capabilities.

Reply:
If you can think of a way to express it, it would be a simple way to future-proof it a bit. It is definitely not a blocker.